staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -720,6 +729,102 @@ func ValidateCustomResourceColumnDefinition(col *apiextensions.CustomResourceCol
 	return allErrs
 }
 
+func validateFieldPath(fieldPath string, pathOfFieldPath *field.Path, schema *apiextensions.JSONSchemaProps) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(fieldPath) == 0 {
+		allErrs = append(allErrs, field.Invalid(pathOfFieldPath, fieldPath, "must not be empty"))
+		return allErrs
+	}
+
+	nonexistentFieldError := field.Invalid(pathOfFieldPath, fieldPath, "does not refer to an existent field")
+
+	var s scanner.Scanner
+	s.Init(strings.NewReader(fieldPath))
+	s.Filename = pathOfFieldPath.String()
+	s.Mode = scanner.ScanInts | scanner.ScanIdents | scanner.ScanChars
+	s.Error = func(s *scanner.Scanner, msg string) {}
+
+	found := false
+	for true {
+		tok := s.Scan()
+		if tok == scanner.EOF {
+			found = true
+			break
+		}
+		switch schema.Type {
+		case "object":
+			if s.TokenText() != "." {
+				allErrs = append(allErrs, nonexistentFieldError)
+				return allErrs
+			}
+			tok = s.Scan()
+			if tok == scanner.EOF {
+				allErrs = append(allErrs, nonexistentFieldError)
+				return allErrs
+			}
+			if schema.Properties != nil {
+				propertySchema, ok := schema.Properties[s.TokenText()]
+				if ok {
+					schema = &propertySchema
+				} else {
+					allErrs = append(allErrs, nonexistentFieldError)
+					return allErrs
+				}
+			} else if schema.AdditionalProperties != nil {
+				additionalProperty, ok := schema.AdditionalProperties.Schema.Properties[s.TokenText()]
+				if ok {
+					schema = &additionalProperty
+				} else {
+					allErrs = append(allErrs, nonexistentFieldError)
+					return allErrs
+				}
+			} else {
```

Comment:
I was expecting the AdditionalProperties case to check for "[" and "]" like the array case.  E.g `.foo.map[<string literal map key>]` where quoting of string literals should match whatever the JSON Path specification requires.

This case needs tests.